docs/docker/intro.md

```diff
@@ -0,0 +1,33 @@
+# Docker
+
+Docker is a container based virtualization software that allow it's user to create a development environment in a easy way without installing a virtualization software i.e VirtualBox.
+
+This allow developer to create multiple test environment on the fly and to share with others without too much headaches.
+
+# Architecture
+
+Docker architecture is centralized on the daemon. The daemon is used by all of the component of Docker. 
+
+In order to control all of these components Docker is using a rest api endpoint that is used by the docker cli. This CLI will send request to the api endpoint which will then be analyze by the docker daemon.
+
+An interesting article regarding this topic can be founded here: [Communicate with the Docker rest API](https://blog.trifork.com/2013/12/24/docker-from-a-distance-the-remote-api/)
+
+<p align="center"> 
+  <img src="../img/docker.png" alt="drawing" width="400"/>
+</p>
+<p align="center"><b>Docker architecture. Docker documentation</b></p>
+
+# Components
+
+In order to use Docker we need to know the several components that has Docker.
```

Comment:
In order to use Docker, we need to know what components it has.

Reply:
Done